### ISSUE TITLE
debug.keystore is not sufficient key size

### ISSUE THREAD

**Issue by STAR-ZERO** [DroidKaigi/conference-app-2020#382]
**Overview (Required)**
- I got below error when using bundletool.

```
[BT:0.12.0] Error: Failed to write APK file '/var/folders/wz/p596hjj11cjf55pj9bm6v1pr0000gn/T/7848660085161609879/splits/base-xxhdpi.apk'.
java.io.UncheckedIOException: Failed to write APK file '/var/folders/wz/p596hjj11cjf55pj9bm6v1pr0000gn/T/7848660085161609879/splits/base-xxhdpi.apk'.
        at com.android.tools.build.bundletool.io.ConcurrencyUtils.waitFor(ConcurrencyUtils.java:55)
        at com.android.tools.build.bundletool.io.ConcurrencyUtils.waitForAll(ConcurrencyUtils.java:42)
        at java.util.function.Function.lambda$andThen$1(Function.java:88)
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:570)
        at com.android.tools.build.bundletool.io.ApkSerializerManager.serializeApks(ApkSerializerManager.java:181)
        at com.android.tools.build.bundletool.io.ApkSerializerManager.populateApkSetBuilder(ApkSerializerManager.java:101)
        at com.android.tools.build.bundletool.commands.BuildApksManager.executeWithZip(BuildApksManager.java:233)
        at com.android.tools.build.bundletool.commands.BuildApksManager.execute(BuildApksManager.java:117)
        at com.android.tools.build.bundletool.commands.BuildApksCommand.execute(BuildApksCommand.java:532)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:74)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:46)
Caused by: java.io.IOException: Failed to generate v1 signature
        at com.android.tools.build.apkzlib.sign.SigningExtension.onOutputZipReadyForUpdate(SigningExtension.java:287)
        at com.android.tools.build.apkzlib.sign.SigningExtension.access$200(SigningExtension.java:57)
        at com.android.tools.build.apkzlib.sign.SigningExtension$1.lambda$beforeUpdate$2(SigningExtension.java:156)
        at com.android.tools.build.apkzlib.zip.ZFile.notify(ZFile.java:2269)
        at com.android.tools.build.apkzlib.zip.ZFile.update(ZFile.java:1002)
        at com.android.tools.build.apkzlib.zip.ZFile.close(ZFile.java:1335)
        at com.android.tools.build.bundletool.io.ApkSerializerHelper.writeToZipFile(ApkSerializerHelper.java:196)
        at com.android.tools.build.bundletool.io.ApkSerializerHelper.writeToZipFile(ApkSerializerHelper.java:139)
        at com.android.tools.build.bundletool.io.SplitApkSerializer.writeToDisk(SplitApkSerializer.java:75)
        at com.android.tools.build.bundletool.io.SplitApkSerializer.writeSplitToDisk(SplitApkSerializer.java:53)
        at com.android.tools.build.bundletool.io.ApkSetBuilderFactory$ApkSetArchiveBuilder.addSplitApk(ApkSetBuilderFactory.java:105)
        at com.android.tools.build.bundletool.io.ApkSerializerManager$ApkSerializer.serialize(ApkSerializerManager.java:377)
        at com.android.tools.build.bundletool.io.ApkSerializerManager.lambda$null$3(ApkSerializerManager.java:185)
        at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:117)
        at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:38)
        at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:77)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.security.InvalidKeyException: Failed to sign using signer "CERT"
        at com.android.apksig.internal.apk.v1.V1SchemeSigner.signManifest(V1SchemeSigner.java:295)
        at com.android.apksig.internal.apk.v1.V1SchemeSigner.sign(V1SchemeSigner.java:256)
        at com.android.apksig.DefaultApkSignerEngine.outputJarEntries(DefaultApkSignerEngine.java:424)
        at com.android.tools.build.apkzlib.sign.SigningExtension.onOutputZipReadyForUpdate(SigningExtension.java:285)
        ... 18 more
Caused by: java.security.InvalidKeyException: Failed to sign using SHA1withDSA
        at com.android.apksig.internal.apk.v1.V1SchemeSigner.generateSignatureBlock(V1SchemeSigner.java:519)
        at com.android.apksig.internal.apk.v1.V1SchemeSigner.signManifest(V1SchemeSigner.java:293)
        ... 21 more
Caused by: java.security.InvalidKeyException: The security strength of SHA-1 digest algorithm is not sufficient for this key size
        at sun.security.provider.DSA.checkKey(DSA.java:111)
        at sun.security.provider.DSA.engineInitSign(DSA.java:143)
        at java.security.Signature$Delegate.init(Signature.java:1155)
        at java.security.Signature$Delegate.chooseProvider(Signature.java:1115)
        at java.security.Signature$Delegate.engineInitSign(Signature.java:1179)
        at java.security.Signature.initSign(Signature.java:530)
        at com.android.apksig.internal.apk.v1.V1SchemeSigner.generateSignatureBlock(V1SchemeSigner.java:515)
        ... 22 more
```

**Steps To Reproduce**

Run below commands.

```
$ ./gradlew clean :android-base:bundleDebug
$ java -jar bundletool-all.jar build-apks \
             --bundle=android-base/build/outputs/bundle/debug/android-base-debug.aab \
             --output=android-base/build/outputs/bundle/debug/android-base-debug.apks \
             --ks=android-base/debug.keystore \
             --ks-pass=pass:android \
             --ks-key-alias=androiddebugkey \
             --key-pass=pass:android \
             --connected-device
```

**Expected behavior**
No error, and create apks fils.

**Screenshots**

**Env:**

**Additional context**

- Using `~/.android/debug.keystore` which is on my computer doesn't occur the error.

**Comment by sdsd08013** [comment]
🙋‍♂

**Comment by FujiKinaga** [comment]
@sdsd08013 
Thanks! Assigned 👍

**Comment by FujiKinaga** [comment]
It seems to be fixed by #536 so I'll close now!
